index.js

```diff
@@ -72,30 +72,40 @@ module.exports = function(func, timeout, options) {
             return func.apply(null, args);
 
         var key = resolver.apply(null, args);
-        var value = null;
         var clear = null;
         var timer = null;
+        var applyTimeout = null;
+
+        function cancelTimeout() {
+            if (timer) {
+                clearTimeout(timer);
+                timer = null;
+            }
+        }
 
         // Populate the cache when there is nothing there yet.
         if (!cache.has(key)) {
-            value = func.apply(null, args);
-            clear = function() { cache.delete(key); };
-            timer = setTimeout(clear, timeout);
-
             var cacheItem = {
                 key: key,
-                value: value,
-                clear: clear,
```

Comment:
Would prefer adding `setTimeout/ttl` and `clear` the way it used to be here.
Including a comment for `eslint` allowing `use-before-define` makes this much easier to understand.